### ISSUE TITLE
Native.load(File) is not accessible due missing Export-Package

### ISSUE THREAD

**Issue by bjmi** [luben/zstd-jni#129]
Calling `Native.load(File)` as described in this [comment](https://github.com/luben/zstd-jni/issues/102#issuecomment-514268136)  is not possible because package _com.github.luben.zstd.util_ is not exported in _MANIFEST.MF_.

The error is described as following:

> Access restriction: The type 'Native' is not API (restriction on required library 'c:\workspace\.metadata\.plugins\org.eclipse.pde.core\.bundle_pool\plugins\com.github.luben.zstd-jni_1.4.5.6.jar')

Additionally please consider adding this header in _MANFEST.MF_. 
`Eclipse-BundleShape: dir`

This allows loading libraries directly because the bundle is already extracted in filesystem.
Reference: https://help.eclipse.org/2020-06/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fmisc%2Fbundle_manifest.html

**Comment by luben** [comment]
Can you double check the latest commit. I think it should fix the issue. Here is the generated manifest (though not including all platforms):

```
Manifest-Version: 1.0
Automatic-Module-Name: com.github.luben.zstd_jni
Bnd-LastModified: 1597920893823
Bundle-Description: JNI bindings for Zstd native library that provides
  fast and high compression lossless algorithm for Java and all JVM la
 nguages.
Bundle-License: https://opensource.org/licenses/BSD-2-Clause;descripti
 on=BSD 2-Clause License
Bundle-ManifestVersion: 2
Bundle-Name: zstd-jni
Bundle-SymbolicName: com.github.luben.zstd-jni
Bundle-Vendor: com.github.luben
Bundle-Version: 1.4.5.7
Created-By: 1.8.0_265 (Private Build)
Eclipse-BundleShape: dir
Export-Package: com.github.luben.zstd;version="1.4.5.7",com.github.lub
 en.zstd.util;version="1.4.5.7"
Implementation-Title: zstd-jni
Implementation-Vendor: com.github.luben
Implementation-Vendor-Id: com.github.luben
Implementation-Version: 1.4.5-7
Import-Package: com.github.luben.zstd.util
Private-Package: include,linux.amd64
Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.8))"
Specification-Title: zstd-jni
Specification-Vendor: com.github.luben
Specification-Version: 1.4.5-7
Tool: Bnd-4.0.0.201805111645
```

Does it look right? I don't have much experience with OSGI bundles TBH.

**Comment by bjmi** [comment]
It looks right but now I noticed that a copy of the library is still copied to the temp folder and is loaded there although the complete bundle lies extracted in the filesystem.

I.e. there is no use declaring `Eclipse-BundleShape: dir`. Exporting _com.github.luben.zstd.util_ suffices. I apologize for the inconvenience.

My workaround in an eclipse application would be then
* Access private method `Native.resourceName()` reflectivly
* Let the runtime extract the library to filesystem (with the help of `org.eclipse.core.runtime.FileLocator.toFileURL(URL)`) and determine absolute path
* Call `java.lang.System.load(String)` by myself
* Set `Native.loaded` to true reflectivly

**Comment by bjmi** [comment]
It is more difficult than expected.

Running in Eclipse `Class.getResource(resourceName)` returns an URL with _bundleentry_ protocol e.g. `bundleentry://9.fwk309775287/win/amd64/libzstd-jni.dll` even if the bundle lies extracted in file system. Therefore checking for _file_ protocol doesn't work in this case. I'll open a new issue if I find a solution for that.

Returning to initial request: just export _com.github.luben.zstd.util_ package.
Thanks in advance.

**Comment by luben** [comment]
addressed and released in v1.4.5-7
